SignalServiceKit/src/Protos/ProtoWrappers/SSKEnvelope.swift

```diff
@@ -0,0 +1,149 @@
+//
+//  Copyright (c) 2018 Open Whisper Systems. All rights reserved.
+//
+
+import Foundation
+
+@objc
+public class SSKEnvelope: NSObject {
+
+    enum EnvelopeError: Error {
+        case invalidProtobuf(description: String)
+    }
+
+    @objc
+    public enum SSKEnvelopeType: Int32 {
+        case unknown = 0
+        case ciphertext = 1
+        case keyExchange = 2
+        case prekeyBundle = 3
+        case receipt = 5
+    }
+
+    @objc
+    public let timestamp: UInt64
+
+    @objc
+    public let source: String
+
+    @objc
+    public let sourceDevice: UInt32
+
+    @objc
+    public let type: SSKEnvelopeType
+
+    @objc
+    public let relay: String?
+
+    @objc
+    public let content: Data?
+
+    @objc
+    public let legacyMessage: Data?
+
+    @objc
+    public init(timestamp: UInt64, source: String, sourceDevice: UInt32, type: SSKEnvelopeType, content: Data?, legacyMessage: Data?) {
+        self.source = source
+        self.type = type
+        self.timestamp = timestamp
+        self.sourceDevice = sourceDevice
+        self.relay = nil
+        self.content = content
+        self.legacyMessage = legacyMessage
+    }
```

Comment:
Shouldn't we explicitly invoke `super.init()` in these initializers?

Reply:
Swift does this implicitly as necessary. You only need to do it explicitly if you want to do something *after* the super initializer.

In general, Swift makes it pretty hard to do the wrong thing with initializers.